imquality/brisque.py

```
@@ -42,7 +42,10 @@ def __init__(
         sigma: float = 7 / 6,
     ):
         self.image = pil2ndarray(image)
-        self.image = skimage.color.rgb2gray(self.image)
+
+        if self.image.shape[-1] == 3:
```

There are images with 4 channels. Do you think that we can generalize this solution to those too?

---

Hi, this is affecting us as well.

Regarding the images with 4 channels, that could be an issue as well, but the thing is that the call to `skimage.color.rgb2gray` will eventually check the same condition. See https://github.com/scikit-image/scikit-image/blob/61ddd88a9f1d68920bf85c0c6f37e5fe69c07c9f/skimage/color/colorconv.py#L137

Hence guarding this call with this condition makes sense.

---

@ocampor Thinking a bit more about this issue and the fact that you are using `rgb2gray` to convert an RGB image to Gray, it makes me think that somehow your code should be more explicit about using RGB images.

For instance, in https://github.com/ocampor/image-quality/blob/d548cdedb9e3779b553ff11e12cfcd28c4a5273f/imquality/brisque.py#L160 you could force the PIL image to be in RGB format:
`image = image.convert('RGB')`

Alternatively, you could raise an exception if the image is not an RGB so the user takes care of this, but this wouldn't be backwards compatible.

Anyways, enforcing the RGB format I believe you can always make sure that the image has 3 channels. There is more information regarding the PIL supported formats in here: https://pillow.readthedocs.io/en/stable/handbook/concepts.html

---

@njay225 could you add the changes requested by @HodeiG ?